brave/app/scripts/controllers/bitgo.js

```diff
@@ -0,0 +1,197 @@
+const ObservableStore = require('obs-store')
+const bgUtil = require('brave-bitgo-client').util
+
+const supportedCoins = {
+  btc: 'Bitcoin',
+  bch: 'Bitcoin Cash',
+  btg: 'Bitcoin Gold',
+  bsv: 'Bitcoin SV',
+  zec: 'ZCash',
+  xlm: 'Stellar',
+  xrp: 'Ripple',
+  ltc: 'Litecoin',
+  dash: 'Dash',
+  algo: 'Algorand',
+  trx: 'Tron',
+  eos: 'Eos'
+}
+
+const uint8ToArrayBuf = (array) => {
+  return array.buffer.slice(array.byteOffset, array.byteLength + array.byteOffset)
+}
+
+const promiseGetSeed = (key) => {
+  return new Promise((resolve) => {
+    if (window.chrome && window.chrome.braveWallet) {
+      window.chrome.braveWallet.getBitGoSeed(key, resolve)
+    } else {
+      // polyfill
+      const seed = new Uint8Array(32)
+      window.crypto.getRandomValues(seed)
+      resolve(uint8ToArrayBuf(seed))
+    }
+  })
+}
+
+class BitGoController {
+  constructor (opts = {}) {
+    const initState = opts.initState || {}
+    this.proxyOrigin = opts.isProduction
+      ? 'https://bitgo-proxy.brave.com/'
+      : 'http://localhost:3000/'
+    this.store = new ObservableStore(initState)
+    this.keyringController = opts.keyringController
+    chrome.braveWallet.getProjectID((projectId) => {
```

Comment:
it's possible this callback won't be called before the user tries to make a request, so the request will fail. might not be worth the effort of handling, but i tried to address this originally by making the constructor-caller call getProjectID before this is constructed. so https://github.com/brave/ethereum-remote-client/pull/136/files#diff-67bab0e473881066a4a1aa0f340934fcR10-R13 would be 

```
chrome.braveWallet.getProjectID((p) => {
  this.bitGoController = new BitGoController(...)
})

Reply:
i've made this change